### ISSUE TITLE
A problem with destructuring in variable declarations

### ISSUE THREAD

**Issue by Amphiluke** [systemjs/builder#541]
Hello,

I faced the challenge at building a module which applies ES6 destructuring to declare variables.
Here's a minimum setup to reproduce the issue

#### The module itself (`src/module.js`):

```javascript
define(function () {
    let obj = {a: 1, b: 2, c: 3};
    let {a, b} = obj;
    return {prop: a + b, prop2: b * a};
});
```

#### Build script:

```javascript
var builder = new Builder("./src");
return builder.bundle("module.js", "build/module.js");
```

For some reason build fails throwing the following exception:

> Error on instantiate for module.js at file:///C:/test/src/module.js
    TypeError: Cannot read property 'value' of undefined
at ParseTreeTransformer.AMDDependenciesTransformer.transformVariableDeclaration (C:\test\node_modules\systemjs-builder\compilers\amd.js:47:34)
at VariableDeclaration.$__super.transform (C:\test\node_modules\traceur\bin\traceur.js:8903:28)
at ParseTreeTransformer.transformAny (C:\test\node_modules\traceur\bin\traceur.js:9514:44)
at ParseTreeTransformer.transformList (C:\test\node_modules\traceur\bin\traceur.js:9521:34)
at ParseTreeTransformer.transformVariableDeclarationList (C:\test\node_modules\traceur\bin\traceur.js:10380:33)
at VariableDeclarationList.$__super.transform (C:\test\node_modules\traceur\bin\traceur.js:8922:28)
at ParseTreeTransformer.transformAny (C:\test\node_modules\traceur\bin\traceur.js:9514:44)
at ParseTreeTransformer.transformVariableStatement (C:\test\node_modules\traceur\bin\traceur.js:10387:33)
at VariableStatement.$__super.transform (C:\test\node_modules\traceur\bin\traceur.js:8940:28)
at ParseTreeTransformer.transformAny (C:\test\node_modules\traceur\bin\traceur.js:9514:44)
at ParseTreeTransformer.transformList (C:\test\node_modules\traceur\bin\traceur.js:9521:34)
at ParseTreeTransformer.transformFunctionBody (C:\test\node_modules\traceur\bin\traceur.js:9895:31)
at FunctionBody.$__super.transform (C:\test\node_modules\traceur\bin\traceur.js:7592:28)
at ParseTreeTransformer.transformAny (C:\test\node_modules\traceur\bin\traceur.js:9514:44)
at ParseTreeTransformer.transformFunctionExpression (C:\test\node_modules\traceur\bin\traceur.js:9917:25)
at ParseTreeTransformer.AMDDependenciesTransformer.transformFunctionExpression (C:\test\node_modules\systemjs-builder\compilers\amd.js:61:69)

The strange thing here is that build succeeds if destructuring is applied outside the variable declaration statement:

```javascript
define(function () {
    let obj = {a: 1, b: 2, c: 3};
    let a, b;
    ({a, b} = obj);
    return {prop: a + b, prop2: b * a};
});
```

**Comment by guybedford** [comment]
Thanks for the clear description here. Fixed in https://github.com/systemjs/builder/commit/5c5b744c8b16a9cd580bd09ab1b56743c8b1610e for the next release. Shouldn't be too hard to patch in now if you need it sooner though.

**Comment by guybedford** [comment]
Released in 0.15.15.
